packages/flutter_tools/lib/src/isolated/devtools_launcher.dart

```diff
@@ -15,47 +19,94 @@ import '../resident_runner.dart';
 /// a devtools dep in google3.
 class DevtoolsServerLauncher extends DevtoolsLauncher {
   DevtoolsServerLauncher({
+    @required ProcessManager processManager,
+    @required String pubExecutable,
     @required Logger logger,
-  }) : _logger = logger;
+  })  : _processManager = processManager,
+        _pubExecutable = pubExecutable,
+        _logger = logger;
 
+  final ProcessManager _processManager;
+  final String _pubExecutable;
   final Logger _logger;
 
-  io.HttpServer _devtoolsServer;
+  io.Process _devToolsProcess;
+  Uri _devToolsUri;
+
+  static final RegExp _serveDevToolsPattern =
+      RegExp(r'Serving DevTools at ((http|//)[a-zA-Z0-9:/=_\-\.\[\]]+)');
 
   @override
-  Future<void> launch(Uri observatoryAddress) async {
+  Future<void> launch(Uri vmServiceUri, {bool openInBrowser = false}) async {
+    if (_devToolsProcess != null && _devToolsUri != null) {
+      // DevTools is already running.
+      if (openInBrowser) {
+        await Chrome.start(<String>[_devToolsUri.toString()]);
+      }
+      return;
+    }
+
     try {
-      await serve();
-      await devtools_server.launchDevTools(
-        <String, dynamic>{
-          'reuseWindows': true,
-        },
-        observatoryAddress,
-        'http://${_devtoolsServer.address.host}:${_devtoolsServer.port}',
-        false,  // headless mode,
-        false,  // machine mode
-      );
+      final io.ProcessResult _devToolsActivateProcess = await _processManager.run(<String>[
+        _pubExecutable,
+        'global',
+        'activate',
+        'devtools'
+      ]);
+      if (_devToolsActivateProcess.exitCode != 0) {
+        _logger.printError('Error running `pub global activate '
+            'devtools`. Could not activate devtools.');
```

Comment:
Could the stdout/stderr here help the end-user resolve the issue? If so, it would probably be good to print it.